app/test_main.py

```diff
@@ -1 +1,35 @@
-# write your code here
+from typing import Any, Type
+from unittest.mock import patch
+from app.main import can_access_google_page
+from pytest import fixture
+
+
+@fixture
+def manager_fixture() -> Any:
+    with (patch("app.main.valid_google_url") as valid_url,
+          patch("app.main.has_internet_connection") as has_connection):
+        yield valid_url, has_connection
+
+
+def test_can_access_google_page(manager_fixture: Type[fixture]) -> None:
+    valid_url, has_connection = manager_fixture
+    valid_url.return_value = True
+    has_connection.return_value = True
+    action = can_access_google_page("https://www.google.com/")
+    assert action == "Accessible"
+
+
+def test_no_access_if_only_connection(manager_fixture: Type[fixture]) -> None:
+    valid_url, has_connection = manager_fixture
+    valid_url.return_value = False
+    has_connection.return_value = True
+    action = can_access_google_page("https://www.google.com/")
+    assert action == "Not accessible"
+
+
+def test_no_access_if_only_url(manager_fixture: Type[fixture]) -> None:
+    valid_url, has_connection = manager_fixture
+    valid_url.return_value = True
+    has_connection.return_value = False
+    action = can_access_google_page("https://www.google.com/")
+    assert action == "Not accessible"
```

Comment:
could u use parametrize here to make ur code DRY ?
also what about False/False case ?